src/Hamelin/Internal/PipelineHost.cs

```diff
@@ -49,7 +52,25 @@ private async Task RunPipeline(CancellationToken cancellationToken)
                 return;
             }
 
-            await step.Run(cancellationToken);
+            try
+            {
+                await step.Run(cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                switch (options.Value.TerminationMode)
+                {
+                    case PipelineTerminationMode.StopOnUnhandledException:
+                        logger.LogCritical(ex, "Unhandled error during pipeline execution.");
+                        lifetime.StopApplication();
+                        throw;
+                    case PipelineTerminationMode.StopAfterAllSteps:
+                        logger.LogError(ex, "Unhandled error during pipeline execution.");
+                        break;
+                    default:
+                        throw;
```

Comment:
Since `StopOnUnhandledException` is the default bahaviour, can we move its handling to the `default` case and fallthrough to there? Just in case something unorthodox happens, it means there's always an explicit log event.

Also added slight differentiation to the log messages to indicate to the user that what happens next is expected behaviour 😄

```suggestion
                    case PipelineTerminationMode.StopAfterAllSteps:
                        logger.LogError(ex, "Unhandled error during pipeline execution. Continuing...");
                        break;
                    case PipelineTerminationMode.StopOnUnhandledException: // This is the default behaviour, so fall through to default case
                    default:
                        logger.LogCritical(ex, "Unhandled error during pipeline execution. Exiting.");
                        lifetime.StopApplication();
                        throw;
```